### PR TITLE
Integration/Routing Branch - Route State and Merge Issues

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
The route_generator package was added back incorrectly by the PR below. Only the route package should exist in the integration/routing branch. Please remove the route_generator package, files, and references.
https://github.com/usdot-fhwa-stol/carma-platform/pull/753/files?file-filters%5B%5D=.h&file-filters%5B%5D=.txt&hide-deleted-files=true

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/806

## How Has This Been Tested?
Build carma-platform, carma-web-ui, autoware.ai, carma-msgs, carma-utils, and novatel_gps_driver repos
launch carma_src.launch in source code, and verifying that:
route_generator package should not exist
route_state should always be publishing after select  a route.

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by dan-du-car** [comment]
> # PR Details
> ## Description
> The route_generator package was added back incorrectly by the PR below. Only the route package should exist in the integration/routing branch. Please remove the route_generator package, files, and references.
> https://github.com/usdot-fhwa-stol/carma-platform/pull/753/files?file-filters%5B%5D=.h&file-filters%5B%5D=.txt&hide-deleted-files=true
> 

Removed below includes from : (https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/motion_computation/include/motion_computation_worker.h
and https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/motion_computation/src/motion_computation_worker.cpp)
#include <autoware_msgs/DetectedObject.h>
#include <autoware_msgs/DetectedObjectArray.h> 

update test for route package and remove route_generator package
> ## Related Issue
> ## Motivation and Context
> ## How Has This Been Tested?
> Build carma-platform, carma-web-ui, autoware.ai, carma-msgs, carma-utils, and novatel_gps_driver repos
> launch carma_src.launch in source code, and verifying that:
> route_generator package should not exist
> route_state should always be publishing after select a route.
> 
> ## Types of changes
> * [x]  Defect fix (non-breaking change that fixes an issue)
> * [ ]  New feature (non-breaking change that adds functionality)
> * [ ]  Breaking change (fix or feature that cause existing functionality to change)
> 
> ## Checklist:
> * [ ]  I have added any new packages to the sonar-scanner.properties file
> * [ ]  My change requires a change to the documentation.
> * [ ]  I have updated the documentation accordingly.
> * [ ]  I have read the **CONTRIBUTING** document.
>   [CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md)
> * [ ]  I have added tests to cover my changes.
> * [ ]  All new and existing tests passed.

**Comment by dan-du-car** [comment]
image

**Comment by msmcconnell** [comment]
Never mind. Ignore my last comment. Sudhakar doesn't want us to commit the TFHRC map.

**Comment by dan-du-car** [comment]
> Please update lines 47 and 50 of this script to point to the tfhrc vector and route files.
> https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/engineering_tools/opt_carma_setup.bash
> 
> Be careful to use the raw.githubusercontent.com versions of the file links and test the curl command locally before pushing.
> 
> This way new development installations will have valid configurations.

@msmcconnell Addressed https://github.com/usdot-fhwa-stol/carma-platform/pull/818/commits/f8b75958056005d623fbbd4a16ce6272f2db9ac9

**Comment by dan-du-car** [comment]
For testing purposes , using different initial current vehicle locations with different route files:
For Test_lanelet135_route_1.csv file : current_loc( 45.8562, -267.505 );
For Test_lanelet111_route_1.csv and Test_lanelet111_route_2.csv files : current_loc( 193.7078, -142.049 );

**Review by MishkaMN — APPROVED** [review]

File: route/include/route_state_worker.h [hunk 1]
```diff
@@ -71,6 +71,9 @@ namespace route {
              */
             RouteState get_route_state();
 
+            //convert RouteState state field to unit8_t
+            uint8_t convertRouteStateToInt ( RouteStateWorker::RouteState state ) const;
```
Comment (msmcconnell) [inline route/include/route_state_worker.h:75]: I believe the route state enum should implicitly convert to int which would make this function unneeded.

File: route/src/route_generator_worker.cpp [hunk 2]
```diff
@@ -347,7 +347,8 @@ namespace route {
             state_msg.routeID = route_msg_.route_name;
             state_msg.cross_track = current_crosstrack_distance_;
             state_msg.down_track = current_downtrack_distance_;
-            state_msg.lanelet_downtrack = ll_downtrack_distance_;
+            state_msg.lanelet_downtrack = ll_downtrack_distance_;            
+            state_msg.state = this->rs_worker_.convertRouteStateToInt(this->rs_worker_.get_route_state());
```
Comment (msmcconnell) [inline route/src/route_generator_worker.cpp:351]: You should be able to just make this
```
state_msg.state = this->rs_worker_.get_route_state();
```

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: route/src/route_generator_worker.cpp (hunk 2 above)
Comment (dan-du-car) [reply route/src/route_generator_worker.cpp]: https://github.com/usdot-fhwa-stol/carma-platform/pull/818/commits/cb5fb9be75570720d340a2e0151aed275157cd56

**Pull request merged**